Release 2.5.0

r-i-v-a opened the pull request:
I have tried `git merge origin/main -X ours` on this branch, but there are conflicts that can't be automatically resolved. Also, this strategy for merging to main means that a lot of old commits will always show up in the branch diff, making it hard to see what is part of the new release.

Maybe we should just make release-X.Y.Z branches off of develop and keep them? It would also mean not having to move the git tag from the release branch onto main at the end of a release? @jdidion @commandlinegirl

commandlinegirl committed:
APPS-542 Document how to enable dxCompiler in a new region (#98)

jdidion committed:
call field access is trivial, but not struct

jdidion committed:
Revert "call field access is trivial, but not struct"

This reverts commit 04fb5eeae56e22bded4429efd996c37bd2b27e76.

jdidion committed:
APPS-544 - don't create default value for file-type outputs (#99)

If an output parameter has expression that is a File, Directory or collection thereof, the paths will be local to the worker so we cannot use them as the default value (since file inputs need to be given as links to dx files).

This PR also activates some additional integration tests and removes unused tests.

jdidion committed:
merge

jdidion committed:
Main (#105)

Merge main into develop

jdidion committed:
Merge branch 'develop' of github.com:dnanexus/dxCompiler into develop

jdidion committed:
update version and changelog

jdidion committed:
APPS 516 - error dereferencing struct as a call argument (#104)

No longer classify struct field access as a trivial expression.

commandlinegirl committed:
Create dxCompiler_test_projects_stg.txt (#91)

Commit:
APPS-556 dxCompiler, use dx whoami for folder in tests, add clean build script (#106)

use dx whoami for folder in tests, add clean build script

Commit:
APPS-472 Wait on upload, dxCompiler (#101)

Update dxScala, set wait = true on DxApi.uploadFile

jdidion committed:
APPS-564 - file gets localized to the wrong folder (#107)

Correctly parse dx URI with file path

jdidion committed:
update developer notes

jdidion committed:
update developer notes

jdidion committed:
update release action

jdidion committed:
update developer notes

Commit:
APPS-472 Integration test for -waitOnUpload. APPS-559 dxCompiler, use dx whoami for build folder. (#109)

jdidion committed:
update changelog

jdidion committed:
add test

Commit:
APPS-559 Use dx whoami --id to match folder name. (#111)

jdidion committed:
APPS-573 update wdlTools (#112)

* allow passing test name to clean_build.sh

* update wdlTools version, add test

* update changelog

* update version

* add wdlTools snapshot resolver

* add dxCompiler snapshot repo

* add dxCompiler snapshot repo

Commit:
APPS-579 Add regression tests for expression evaluation: boolean, substitution. (#113)

jdidion committed:
Update dxScala, wdlTools versions (#115)

Move from snapshot to release versions

jdidion committed:
disable population test

jdidion committed:
update dxfuse version

emiloslavsky committed:
Document external container registry details (#116)

Commit:
APPS-565 Applet setup for testing dxCompiler in customer project. (#117)

jdidion committed:
bump version, update changelog

jdidion committed:
APPS-598 Concatenate sub-block names for blocks that contain multiple sub-blocks (#120)

concatenate sub-block names for blocks that contain multiple sub-blocks

jdidion committed:
Update to wdlTools 0.13.0 (#122)

Update to wdlTools 0.13.0. Also implements the ability to parse and type-check WDL in lenient mode, although the option is currently not available to the user.

jdidion committed:
fix merge

jdidion committed:
remove test

jdidion committed:
Apps 612 - resolve fully-qualified names when evaluating closure inputs (#125)

Performs a recursive lookup for fully-qualified names when evaluating closure inputs.

jdidion committed:
check that input files are closed (#124)

Check that input files are in the 'closed' state

jdidion committed:
add test

jdidion committed:
update changelog and version

Commit:
APPS-618 Use detached jobs in example test applet script. (#127)

jdidion committed:
Update to wdlTools 0.13.1 (#128)

update wdlTools version

jdidion committed:
update changelog and version

jdidion committed:
grant networks access - required for detached jobs

jdidion committed:
Specify type aliases when converting from WDL to IR type (#131)

jdidion committed:
Make array nonEmpty=false when item types are optional (#132)

Commit:
APPS-634 Fix example script run command, use same flags as in customer project. (#133)

jdidion committed:
add dependency graph plugin

jdidion committed:
update versions and changelogs

jdidion committed:
update dxScala versions (#136)

* update dxScala versions
* update wdlTools version

Commit:
APPS-640 Clarify documentation example for dxCompiler.jar dxni -apps. (#137)

jdidion committed:
Ensure that we don't get nested optional types (#138)

always use Type.ensureOptional rather than blindly wrapping with TOptional

jdidion committed:
APPS-645: miniwdl corpora tests (#140)

jdidion committed:
ignore corpora

jdidion committed:
Merge branch 'develop' of github.com:dnanexus/dxCompiler into develop

Commit:
APPS-670 Add to documentation about what you'll see in project & applet job scripts when compiling. (#142)

jdidion committed:
handle references between output parameters (#144)

jdidion committed:
Specify OS in InstanceTypeRequest (#143)

emiloslavsky committed:
docs: fix example wf, dxCompiler docker release (#146)

Commit:
APPS-677 Fix so that option -execTree pretty, json prints the workflow tree. (#145)

jdidion committed:
DEVEX-1476 add support for metadata in hints section (#126)

* Recognizes localizationOptional in hints section
* Recognizes DNAnexus-specific hints in hints section
* Adds tags and properties to file uploads if specified in hints

jdidion committed:
Update dxApi - new DxFindDataObjects API (#147)

jdidion committed:
Update README.md

jdidion committed:
update version and release notes

jdidion committed:
APPS-685 make input parameters with default values optional in generated WDL (#153)

jdidion committed:
Upload workflow files (#152)

jdidion committed:
update dxCommon and dxApi versions (#154)

Commit:
APPS-721 Ignore dxCompiler archive tests / tests relying on squashfs on non-Linux systems. (#155)

Co-authored-by: John Didion <github@didion.net>

jdidion committed:
update to latest dxCommon (#156)

jdidion committed:
update dxda and dxfuse versions (#158)

jdidion committed:
APPS-615: improve and enable struct_deref integration test (#159)

jdidion committed:
APPS-684: Add more logging for pinpointing job hangs (#160)

jdidion committed:
update test

Commit:
APPS-658 dxCompiler will annotate the generated applets and workflows with their statically-determined dependencies and add them to details & description. (#130)

jdidion committed:
Fix integration test bugs (#164)

jdidion committed:
Apps 749 - fail on disk exhausted (#163)

jdidion committed:
update version and changelog

mhrvol committed:
APPS-604 cromwell tests (#141)

* APPS-604 added cromwell tests.
* APPS-723 integration tests are run in their own folder (to prevent workflow overrides)

Co-authored-by: jdidion <github@didion.net>

Commit:
APPS-764 Dependency report test files deleted; should be stored in dxCompiler_playground:/test_data instead. (#169)

jdidion committed:
APPS-759 Handle empty scatter (#168)

jdidion committed:
update dxda (#167)

jdidion committed:
Apps 538 - document reorg polling (#100)

jdidion committed:
When uploading files in private variables, ignore any files that are already non-local (#170)

jdidion committed:
reset version to snapshot

jdidion committed:
Merge branch 'develop' of github.com:dnanexus/dxCompiler into develop

jdidion committed:
reset version to snapshot

jdidion committed:
Release 2.4.10 (#171)

jdidion committed:
Revert "Release 2.4.10 (#171)" (#172)

This reverts commit 75c089e94c8184de574f8715c8447f84bd0fae4e.

jdidion committed:
update version and changelog

jdidion committed:
Merge branch 'develop' of github.com:dnanexus/dxCompiler into develop

Commit:
APPS-621 dxCompiler will package platform Docker image dependencies in applet's bundledDepends. (#174)

Commit:
APPS-751 Readme / script for extracting source code from metadata. (#162)

jdidion committed:
use type hints when getting expression inputs (#175)

Commit:
APPS-542 dxCompiler new London region (#176)

Commit:
APPS-783 dxCompiler test applet | Restore deleted Makefile, add ignore reuse, unique folders (#178)

jdidion committed:
APPS-782: provide option to disable compile-time instance type evaluation (#177)

Commit:
APPS-786 Add customer workflow testing to release process (#179)

commandlinegirl committed:
APPS-786 Add link to customer test applet (#181)

jdidion committed:
RELENG-890: update permissions in GHA workflows

jdidion committed:
RELENG-890: update permissions in GHA workflows

jdidion committed:
RELENG-891 update permissions in GHA workflows

jdidion committed:
APPS-788 fix project-wide reuse (#183)

* fix project-wide reuse

* fix tests

Commit:
Set release version.

Commit:
Revert "Set release version."

This reverts commit 3cd432c679b1264f6b675e2a7a6c9f2fbef42567.

Commit:
Update release notes.

Commit:
Set release version, fix script for macOS.

Commit:
Release doc -- update libraries item.

Commit:
Release doc -- update libraries item.

Commit:
Set library versions.

jdidion committed:
Apply fix for evaluation of outputs when one declaration depends on another.

Commit:
Try wdlTools 0.16.1-SNAPSHOT for testing (eventual release should depend on released version).

Commit:
wdlTools version 0.17.0.

jdidion committed:
merge main

jdidion commented:
Fixed. There was really only one conflict due to a file being moved in one branch and deleted in another. I find the value of a main branch questionable. The biggest argument for it that I see is that it is protected so it's hard to delete main where as it's easy to delete tags and release branches.

jdidion committed:
fix merge

r-i-v-a commented:
> Fixed. There was really only one conflict due to a file being moved in one branch and deleted in another. I find the value of a main branch questionable. The biggest argument for it that I see is that it is protected so it's hard to delete main where as it's easy to delete tags and release branches.

It's possible to add a protection rule for a name pattern as well as individual branches, so you could protect all `release-*` branches. https://docs.github.com/en/enterprise-server@3.0/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule

In case we deleted a tag by accident, it would generally be the last commit on the same-named release branch, so not hard to find where to put it back.

The process of merging into main just makes me a bit nervous, since so much history shows up in the diff (the commit ids once applied on main don't match the ones in develop / release branch's history), so it's hard to review, and I don't want to make a mistake while resolving conflicts that changes the record of what was released. @jdidion  @commandlinegirl

r-i-v-a commented:
We should still keep main around & protected, since it records some past releases. @jdidion  @commandlinegirl
